src/cascadia/inc/cppwinrt_utils.h

```diff
@@ -288,4 +288,69 @@ std::vector<wil::com_ptr<T>> SafeArrayToOwningVector(SAFEARRAY* safeArray)
     namespace nameSpace::factory_implementation                                                                   \
     {                                                                                                             \
         BASIC_FACTORY(className);                                                                                 \
-    }\
+    }
+
+#ifdef WINRT_Windows_UI_Xaml_H
+
+inline ::winrt::hstring XamlThicknessToOptimalString(const ::winrt::Windows::UI::Xaml::Thickness& t)
+{
+    if (t.Left == t.Right)
+    {
+        if (t.Top == t.Bottom)
+        {
+            if (t.Top == t.Left)
+            {
+                return ::winrt::hstring{ fmt::format(FMT_COMPILE(L"{}"), t.Left) };
+            }
+            return ::winrt::hstring{ fmt::format(FMT_COMPILE(L"{},{}"), t.Left, t.Top) };
+        }
+        // fall through
+    }
+    return ::winrt::hstring{ fmt::format(FMT_COMPILE(L"{},{},{},{}"), t.Left, t.Top, t.Right, t.Bottom) };
+}
+
+inline ::winrt::Windows::UI::Xaml::Thickness StringToXamlThickness(std::wstring_view padding)
+try
+{
+    uintptr_t count{ 0 };
+    double t[4]{ 0. }; // left, top, right, bottom
+    wchar_t buf[17];
+    for (const auto& token : til::split_iterator{ padding, L',' })
+    {
+        const auto l{ std::min(token.size(), std::extent_v<decltype(buf)> - 1) };
+#pragma warning(suppress : 26459) // You called an STL function '...' with a raw pointer parameter at position '...' that may be unsafe ... (stl.1).
+        std::copy_n(token.data(), l, &buf[0]); // the length of buf is controlled for above
+        til::at(buf, l) = L'\0';
+        til::at(t, count++) = std::wcstod(&buf[0], nullptr);

```

Comment:
I believe this would need to do error checking to filter invalid inputs. `Converters::MaxValueFromPaddingString` above does it. Now that I think about it, you could even merge both functions by creating one that returns a `std::array<double, 4>`. 🤔